dot/peerset/peerset.go

```diff
@@ -469,7 +494,7 @@ func (ps *PeerSet) removeReservedPeers(setID int, peers ...peer.ID) error {
 		if ps.peerState.peerStatus(setID, peerID) == connectedPeer {
 			err := ps.peerState.disconnect(setID, peerID)
 			if err != nil {
-				return err
+				return fmt.Errorf("cannot disconnect peer %s at set %d: %w", peerID, setID, err)
```

Comment:
nit wrap the error context about the peer id and set id in `disconnect` instead of here